src/spec/doc/core-metaprogramming.adoc

```diff
@@ -966,6 +966,25 @@ include::{projectdir}/src/spec/test/CodeGenerationASTTransformsTest.groovy[tags=
 ----
 |=======================================================================
 
+[[xform-MapConstructor]]
+===== @groovy.transform.MapConstructor
+
+The `@MapConstructor` annotation aims at eliminating boilerplate code by generating a map constructor for you. A map
```

Comment:
Yes, MapConstructor was missing from the documentation altogether. I added at least a placeholder but I'll probably try to add the doco in a separate commit and then rework this PR accordingly.